docs/data/material/getting-started/templates/checkout/PaymentForm.tsx

```diff
@@ -1,65 +1,272 @@
+    if (value.length <= 16) {
+      setCardNumber(formattedValue);
+    }
+  };
+
+  const handleCvvChange = (event: { target: { value: string } }) => {
+    const value = event.target.value.replace(/\D/g, '');
+    if (value.length <= 3) {
+      setCvv(value);
+    }
+  };
+
+  const handleExpirationDateChange = (event: { target: { value: string } }) => {
+    const value = event.target.value.replace(/\D/g, '');
+    const formattedValue = value.replace(/(\d{2})(?=\d{2})/, '$1/');
+    if (value.length <= 4) {
+      setExpirationDate(formattedValue);
+    }
+  };
+
   return (
-    <React.Fragment>
-      <Typography variant="h6" gutterBottom>
-        Payment method
-      </Typography>
-      <Grid container spacing={3}>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="cardName"
-            label="Name on card"
-            fullWidth
-            autoComplete="cc-name"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="cardNumber"
-            label="Card number"
-            fullWidth
-            autoComplete="cc-number"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="expDate"
-            label="Expiry date"
-            fullWidth
-            autoComplete="cc-exp"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12} md={6}>
-          <TextField
-            required
-            id="cvv"
-            label="CVV"
-            helperText="Last three digits on signature strip"
-            fullWidth
-            autoComplete="cc-csc"
-            variant="standard"
-          />
-        </Grid>
-        <Grid item xs={12}>
+    <Stack spacing={6} useFlexGap>
+      <FormControl component="fieldset" fullWidth>
+        <RadioGroup
+          aria-label="Payment options"
+          name="paymentType"
+          value={paymentType}
+          onChange={handlePaymentTypeChange}
+          sx={{
+            flexDirection: { sm: 'column', md: 'row' },
+            gap: 2,
+          }}
+        >
+          <Card
+            raised={paymentType === 'creditCard'}
+            sx={{
+              maxWidth: { sm: '100%', md: '50%' },
+              flexGrow: 1,
+              outline: '1px solid',
+              outlineColor:
+                paymentType === 'creditCard' ? 'primary.main' : 'divider',
+              backgroundColor:
+                paymentType === 'creditCard' ? 'background.default' : '',
+            }}
+          >
+            <CardActionArea onClick={() => setPaymentType('creditCard')}>
+              <CardContent sx={{ display: 'flex', alignItems: 'center', gap: 1 }}>
+                <CreditCardRoundedIcon color="primary" fontSize="small" />
+                <Typography variant="subtitle1" fontWeight="medium">
```

Comment:
```suggestion
                <Typography fontWeight="medium">
```

We can reduce the size of the label a bit